test/behaviour/reasoner/verification/SoundnessVerifier.java

```diff
@@ -101,14 +114,46 @@ private boolean canExplanationBeVerified(Explanation explanation) {
+            List<Explanation> explanations = tx.query().explain(explainable.id()).toList();
+            iterate(explanations).forEachRemaining(explanation -> {
                 // This check is valid given that there is no mechanism for recursion termination given by the UX of
                 // explanations, so we do it ourselves
                 if (collectedExplanations.contains(explanation)) return;
                 else collectedExplanations.add(explanation);
                 verifyAnswerAndCollectExplanations(explanation.conditionAnswer(), tx);
                 verifyVariableMapping(answer, explainable, explanation);
             });
+
+            Pair<Conjunction, ConceptMap> deferredBindableConjunction = new Pair<>(explainable.conjunction(), answer.filter(explainable.conjunction().retrieves()));
+            assert !deferredExplanationCount.containsKey(deferredBindableConjunction) ||
+                    deferredExplanationCount.get(deferredBindableConjunction).equals(explanations.size());
+            deferredExplanationCount.put(deferredBindableConjunction, explanations.size());
+        });
+    }
+
+    private void verifyNumberOfExplanations(Transaction tx, TypeQLMatch inferenceQuery, Pair<Conjunction, ConceptMap> deferredBindableConjunction) {
+        Set<BoundConcludable> boundConcludable = BoundConjunction.create(
+                deferredBindableConjunction.first(), mapInferredConcepts(deferredBindableConjunction.second())
+        ).boundConcludables();
+        assert boundConcludable.size() == 1;
+        boundConcludable.forEach(bc -> {
+            bc.concludable().getApplicableRules(tx.concepts(), tx.logic());
+            AtomicInteger numExplanationsExpected = new AtomicInteger();
+
+            materialiser.concludableMaterialisations(bc).forEachRemaining(materialisation -> {
+                bc.concludable().getUnifiers(materialisation.boundConclusion().conclusion().rule()).forEachRemaining(unifier -> {
+                    Optional<Pair<ConceptMap, Unifier.Requirements.Instance>> boundsAndRequirements = unifier.unify(bc.pattern().bounds());
+                    assert boundsAndRequirements.isPresent();
+                    numExplanationsExpected.getAndAdd(unifier.unUnify(materialisation.boundConclusion().bounds(), boundsAndRequirements.get().second()).toSet().size());
+                });
+            });
+            if (deferredExplanationCount.get(deferredBindableConjunction) != numExplanationsExpected.get()) {
+                throw new SoundnessException(String.format(
+                        "Soundness testing found an incorrect number of explanations for query \"%s\" for explainable" +
+                                " \"%s\".\nExplanations expected: %s\nExplanations found: %s",
+                        inferenceQuery, deferredBindableConjunction.first(), numExplanationsExpected, deferredExplanationCount.get(deferredBindableConjunction)
+                ));
+            }
```

Comment:
Add a check that the number of explanations is as expected. We do this as we do the soundness check which queries for the user query and queries the body of each rule used. For a given query find the number of explanations that there should be for each concludable, getting this number from the forward-chaining oracle. Compare this with the number of explanations actually found for that concludable (actually from the explainable that represents that concludable).

Doing this required some extra trickery to find the number of actual explanations that should be present since the materialiser doesn't store an exact equivalent to explanations, it has `concludableMaterialisations` instead.